velox/functions/sparksql/tests/ArrayMaxTest.cpp

```diff
@@ -0,0 +1,174 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <type/Timestamp.h>
+#include <limits>
+#include <optional>
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class ArrayMaxTest : public SparkFunctionBaseTest {
+ protected:
+  template <typename T>
+  std::optional<T> arrayMax(const std::vector<std::optional<T>>& input) {
+    std::vector<std::vector<std::optional<T>>> vec = {input};
+    auto row = makeRowVector({makeNullableArrayVector(vec)});
+    return evaluateOnce<T>("array_max(C0)", row);
+  }
+};
+
+TEST_F(ArrayMaxTest, boolean) {
+  EXPECT_EQ(arrayMax<bool>({true, false}), true);
+  EXPECT_EQ(arrayMax<bool>({true}), true);
+  EXPECT_EQ(arrayMax<bool>({false}), false);
+  EXPECT_EQ(arrayMax<bool>({}), std::nullopt);
+  EXPECT_EQ(arrayMax<bool>({true, false, true, std::nullopt}), true);
+  EXPECT_EQ(arrayMax<bool>({std::nullopt, true, false, true}), true);
+  EXPECT_EQ(arrayMax<bool>({false, false, false}), false);
+  EXPECT_EQ(arrayMax<bool>({true, true, true}), true);
+}
+
+TEST_F(ArrayMaxTest, varchar) {
+  EXPECT_EQ(arrayMax<StringView>({"red"_sv, "blue"_sv}), "red"_sv);
+  EXPECT_EQ(
+      arrayMax<StringView>({std::nullopt, "blue"_sv, "yellow"_sv, "orange"_sv}),
+      "yellow"_sv);
+  EXPECT_EQ(arrayMax<StringView>({}), std::nullopt);
+  EXPECT_EQ(arrayMax<StringView>({std::nullopt}), std::nullopt);
+}
+
+// Test non-inlined (> 12 length) nullable strings.
+TEST_F(ArrayMaxTest, longVarchar) {
+  EXPECT_EQ(
+      arrayMax<StringView>(
+          {"red shiny car ahead"_sv, "blue clear sky above"_sv}),
+      "red shiny car ahead"_sv);
+  EXPECT_EQ(
+      arrayMax<StringView>(
+          {std::nullopt,
+           "blue clear sky above"_sv,
+           "yellow rose flowers"_sv,
+           "orange beautiful sunset"_sv}),
+      "yellow rose flowers"_sv);
+  EXPECT_EQ(arrayMax<StringView>({}), std::nullopt);
+  EXPECT_EQ(
+      arrayMax<StringView>(
+          {"red shiny car ahead"_sv,
+           "purple is an elegant color"_sv,
+           "green plants make us happy"_sv}),
+      "red shiny car ahead"_sv);
+}
+
+TEST_F(ArrayMaxTest, date) {
+  auto D = [](const std::string& dateStr) { return DATE()->toDays(dateStr); };
+  EXPECT_EQ(
+      arrayMax<int32_t>({D("1970-01-01"), D("2023-08-23")}), D("2023-08-23"));
+  EXPECT_EQ(arrayMax<int32_t>({}), std::nullopt);
+  EXPECT_EQ(
+      arrayMax<int32_t>({D("1970-01-01"), std::nullopt}), D("1970-01-01"));
+}
+
+TEST_F(ArrayMaxTest, timestamp) {
+  auto T = [](int64_t micros) { return Timestamp::fromMicros(micros); };
```

Comment:
T -> TS maybe

Reply:
We may need to use lower case. I just changed to `ts`.